### PR TITLE
feat: add basic metrics for sharky

### PR DESCRIPTION
**Pull request by mrekucci**
Introduces basic metrics for sharky as read, write counters and their error counters, shard size counter.

---
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/2759)

### CODE REVIEW

File: pkg/sharky/store.go [hunk 1]
```diff
@@ -48,22 +49,25 @@ func New(basedir string, shardCnt int, limit uint32, datasize int) (*Store, erro
...
+	for i := range store.shards {
+		s, err := store.create(uint8(i), limit, datasize, basedir)
 		if err != nil {
 			return nil, err
 		}
-		sh.shards[i] = s
+		store.shards[i] = s
 	}
-	return sh, nil
+	store.metrics.ShardSize.Set(float64(len(store.shards)))
```
Comment (acud) [inline pkg/sharky/store.go:68]: just to mention, this is not `shard size` but `number of shard`. shard size would be the number of the items in the shard, which would be the `filesize/datasize` and should be updated when the shard grows

**Review by acud — APPROVED** [review]
nice! I have some suggestions

**Comment by acud** [comment]
also missing - you need to register the metrics in the `node.go` file somehow (either localstore should throw the collector when being called from `node.go`, or leak the registration to `node.go`

File: pkg/sharky/store.go (hunk 1 above)
Comment (mrekucci) [reply pkg/sharky/store.go]: Done.

**Comment by mrekucci** [comment]
> also missing - you need to register the metrics in the `node.go` file somehow (either localstore should throw the collector when being called from `node.go`, or leak the registration to `node.go`

Yes, but that would be done in the #2682 PR, since the sharky is instantiated there.

**Review by notanatol — APPROVED** [review]

**Review by zelig — COMMENTED** [review]
> Reviewed 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2759)* status: all files reviewed, 1 unresolved discussion (waiting on @acud and @aloknerurkar)

___
*[pkg/sharky/store.go, line 68 at r1](https://reviewable.io/reviews/ethersphere/bee/2759#-MtItPUsPuy20TKtxGAE-r1-68:-MtUXokC9gPr05OXC43K:b-oaz2hm) ([raw file](https://github.com/ethersphere/bee/blob/0583efe062c331c56a57a938c65d74536aeb5c41/pkg/sharky/store.go#L68)):*
<details><summary><i>Previously, mrekucci wrote…</i></summary><blockquote>

Done.
</blockquote></details>

this is simply `shardCnt` variable

**Review by mrekucci — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/2759)* status: all files reviewed, 1 unresolved discussion (waiting on @acud and @aloknerurkar)

___
*[pkg/sharky/store.go, line 68 at r1](https://reviewable.io/reviews/ethersphere/bee/2759#-MtItPUsPuy20TKtxGAE-r1-68:-MtbSJE2646_R-bgNgpl:b-896fix) ([raw file](https://github.com/ethersphere/bee/blob/0583efe062c331c56a57a938c65d74536aeb5c41/pkg/sharky/store.go#L68)):*
<details><summary><i>Previously, zelig (Viktor Trón) wrote…</i></summary><blockquote>

this is simply `shardCnt` variable
</blockquote></details>

Done.

**Pull request merged**